src/OpenTelemetry/Trace/ActivitySourceAdapter.cs

```diff
@@ -80,7 +84,7 @@ public void Start(Activity activity, ActivityKind kind, ActivitySource source)
 
```

Comment:
Could we achieve this by creating a temporary activity with the given source and use its TraceFlags and IsAllDataRequested information in conjunction with the  tracer sampler information to update the activity correctly?

So something like below in the Start method would give us the information to decide the highest level of sampling for the activity.
```
var tempActivity = source.StartActivity("Test");
var isRecordedByOtherListeners = tempActivity.Recorded;
var isAllDataRequestedByOtherListeners = tempActivity.IsAllDataRequested;
tempActivity.Stop();
```

We would incur the cost of allocating a new activity object. We could move this to the ActivitiySourceAdapater ctor but then if any new listener is added in the application afterwards we wouldn't have the updated information.